configs/example/gem5_library/riscv-demo-board-ubuntu-run-24.04.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+This script utilizes the RiscvDemoBoard to run a simple Ubunutu boot. The script
+will boot the the OS to login before exiting the simulation.
+
+A detailed terminal output can be found in `m5out/board.platform.terminal`.
+
+**Warning:** The RiscvDemoBoard uses the Timing CPU. The boot may take
+considerable time to complete execution.
+`configs/example/gem5_library/x86-ubuntu-run-with-kvm.py` can be referenced as
+an example of booting Ubuntu with a KVM CPU.
+
+Usage
+-----
+
+```
+scons build/ALL/gem5.opt
+./build/ALL/gem5.opt configs/example/gem5_library/riscv-demo-board-ubuntu-run-24.04.py
+```
+"""
+
+import argparse
+
+import m5
+
+from gem5.prebuilt.demo.riscv_demo_board import RiscvDemoBoard
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+
+board = RiscvDemoBoard(is_fs=True)
+
+
+def handle_workend():
+    print("Dump stats at the end of the ROI!")
+
+    m5.stats.dump()
+    print("Reset stats at the end of the ROI!")
+    m5.stats.reset()
```

Comment:
Why do we reset stats at the end or the ROI? Isn't this the end of the simulation?